### PR TITLE
handle cipher and key size for encryption

### PR DESCRIPTION
**Pull request by aschnell**
For https://trello.com/c/B72IxUOM/1294-3-probing-of-cipher-and-recognize-preexisting-pervasive-encryption.

### CODE REVIEW

File: storage/SystemInfo/CmdCryptsetup.cc [hunk 1]
```diff
@@ -119,15 +120,123 @@ namespace storage
...
+	}
+
 	y2mil(*this);
     }
 
 
+    void
+    CmdCryptsetupLuksDump::parse_version1(const vector<string>& lines)
+    {
+	static const regex cipher_name_regex("Cipher name:[ \t]*([^ \t]+)[ \t]*", regex::extended);
```
Comment (wfeldt) [inline storage/SystemInfo/CmdCryptsetup.cc]: And also in all regexps below: the terminating "[ \t]*" part has no effect whatsoever.

**Comment by wfeldt** [comment]
Basically, lgtm but I would simplify the regexps.

NP: why store the key size in bytes when it's used everywhere in bits?

File: storage/SystemInfo/CmdCryptsetup.cc (hunk 1 above)
Comment (aschnell) [reply storage/SystemInfo/CmdCryptsetup.cc]: No, regex_match tries to match the complete string. When using regex_search it would be indeed not needed.

**Comment by aschnell** [comment]
Since the on-disk format of LUKS2 also stores the key size in bytes I have implemented it the same way now.

File: storage/SystemInfo/CmdCryptsetup.cc (hunk 1 above)
Comment (wfeldt) [reply storage/SystemInfo/CmdCryptsetup.cc]: Talked to Arvin; the regexps are correct as-is.

**Comment by wfeldt** [comment]
lgtm

**Pull request merged**